engine/crates/parser-sdl/src/rules/mongodb_directive/model_directive/types/filter.rs

```diff
@@ -53,7 +53,16 @@ pub(crate) fn register_type_input<'a>(
     extra_fields: impl Iterator<Item = MetaInputValue> + 'a,
 ) {
     let explicit_fields = object.fields.iter().map(|field| {
-        let r#type = if field.r#type().base.is_list() {
+        let r#type = if matches!(
+            visitor_ctx.types[field.r#type().base.to_base_type_str()].node.kind,
+            engine_parser::types::TypeKind::Enum(_),
+        ) {
+            field.r#type().base.to_string()
+        } else if matches!(
+            visitor_ctx.types[field.r#type().base.to_base_type_str()].node.kind,
+            engine_parser::types::TypeKind::Scalar
+        ) && field.r#type().base.is_list()
```

Comment:
This is pretty obviously wrong. But more correct than what we have, tests are passing. Will try to figure out something principled if we go ahead.